components/brave_wallet/browser/keyring_controller.cc

```diff
@@ -675,6 +720,107 @@ std::vector<mojom::AccountInfoPtr> KeyringController::GetAccountInfosForKeyring(
   return result;
 }
 
+void KeyringController::GetHardwareAccounts(
+    GetHardwareAccountsCallback callback) {
+  std::vector<mojom::HardwareWalletAccountPtr> accounts;
+  base::Value hardware_keyrings(base::Value::Type::DICTIONARY);
+  const base::Value* value = GetPrefForHardwareKeyring(prefs_);
+  if (!value) {
+    std::move(callback).Run({});
+    return;
+  }
+
+  for (const auto hw_keyring : value->DictItems()) {
+    std::string device_id = hw_keyring.first;
```

Comment:
we need to pass back this hardware_id to UI so UI can group same device together

Reply:
as discussed created new issue for this https://github.com/brave/brave-browser/issues/18165